tests/test_models/test_nn/test_deepstate.py

```diff
@@ -47,6 +51,53 @@ def test_deepstate_model_run_weekly_overfit_with_scaler(ts_dataset_weekly_functi
     assert mae(ts_test, future) < 0.001
 
 
+@pytest.mark.parametrize("df_name", ["example_make_samples_df", "example_make_samples_df_int_timestamp"])
+def test_deepstate_make_samples(df_name, request):
+    df = request.getfixturevalue(df_name)
+    ssm = MagicMock()
+    datetime_index = np.arange(len(df))
+    ssm.generate_datetime_index.return_value = datetime_index[np.newaxis, :]
+    module = MagicMock(ssm=ssm)
+    encoder_length = 8
+    decoder_length = 4
+
+    ts_samples = list(
+        DeepStateNet.make_samples(module, df=df, encoder_length=encoder_length, decoder_length=decoder_length)
+    )
+    first_sample = ts_samples[0]
+    second_sample = ts_samples[1]
+
+    assert len(ts_samples) == len(df) - encoder_length - decoder_length + 1
+
+    df["datetime_index"] = datetime_index
```

Comment:
Maybe we should use loops here instead

Reply:
Could you explain about for loops? I didn't understand.

Reply:
you have copy paste below

Reply:
Fixed it